### PR TITLE
DOC: fix napoleon, render sphinx docs right

### PR DESCRIPTION
**Pull request by mikofski**
pvlib python pull request guidelines
====================================

Thank you for your contribution to pvlib python! You may delete all of these instructions except for the list below.

You may submit a pull request with your code at any stage of completion.

The following items must be addressed before the code can be merged. Please don't hesitate to ask for help if you're unsure of how to accomplish any of the items below:

 - [x] Closes issue #690 
 - [x] I am familiar with the [contributing guidelines](http://pvlib-python.readthedocs.io/en/latest/contributing.html).
 - [x] Fully tested. Added and/or modified tests to ensure correct behavior for all reasonable inputs. Tests (usually) must pass on the TravisCI and Appveyor testing services.
 - [ ] Updates entries to `docs/sphinx/source/api.rst` for API changes.
 - [ ] Adds description and name entries in the appropriate `docs/sphinx/source/whatsnew` file for all changes.
 - [x] Code quality and style is sufficient. Passes LGTM and SticklerCI checks.
 - [x] New code is fully documented. Includes sphinx/numpydoc compliant docstrings and comments in the code where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.

Brief description of the problem and proposed solution (if not already fully described in the issue linked to above):

* add napoleon extension to sphinx source `conf.py`
* fix see also in `solar_zenith_analytical`

### CODE REVIEW

**Comment by wholmgren** [comment]
Thanks Mark. My understanding is that we should use either the `napoleon` extension or the `numpydoc` extension, but not both. I googled around a bit for pros/cons of napoleon with numpydoc formatting vs. just using numpydoc. I didn't find anything conclusive. It seems that switching from numpydoc to napoleon is trivial for some projects and difficult for others. We'd need to thoroughly check the build to make sure that all pages render correctly.

**Comment by mikofski** [comment]
new docs as rendered are [here](https://fix-see-also-docs-solar-zenith-analytical.readthedocs.io/en/fix_see_also_docs_solar_zenith_analytical/)

**Comment by wholmgren** [comment]
Overall, I prefer the napoleon formatting. I also like that it links to the allowed types. I wonder if there's a way to link to custom definitions for "numeric" and "array-like" (not required here, just thinking).

I've listed a handful of issues below. Many of these are also issues in the numpydoc build, but the differences in formatting have made them breaking issues instead of nice to have fixes.

* [Location.get_solarposition](https://fix-see-also-docs-solar-zenith-analytical.readthedocs.io/en/fix_see_also_docs_solar_zenith_analytical/generated/pvlib.location.Location.get_solarposition.html#pvlib.location.Location.get_solarposition) param kwargs
* [new line needed after return type in solar_zenith_analytical](https://github.com/pvlib/pvlib-python/blob/master/pvlib/solarposition.py#L1281)
* [Location.get_clearsky](https://fix-see-also-docs-solar-zenith-analytical.readthedocs.io/en/fix_see_also_docs_solar_zenith_analytical/generated/pvlib.location.Location.get_clearsky.html) kwargs spec is not sensible with napoleon. It's just poorly formatted with numpydoc. See [PVSystem.get_irradiance](https://fix-see-also-docs-solar-zenith-analytical.readthedocs.io/en/fix_see_also_docs_solar_zenith_analytical/generated/pvlib.pvsystem.PVSystem.get_irradiance.html#pvlib.pvsystem.PVSystem.get_irradiance) for working kwargs spec.
* [sapm](https://fix-see-also-docs-solar-zenith-analytical.readthedocs.io/en/fix_see_also_docs_solar_zenith_analytical/generated/pvlib.pvsystem.sapm.html) returns section
* [SingleAxisTracker](https://fix-see-also-docs-solar-zenith-analytical.readthedocs.io/en/fix_see_also_docs_solar_zenith_analytical/generated/pvlib.tracking.SingleAxisTracker.html) params not rendered right.
* [ModelChain.prepare_inputs](https://fix-see-also-docs-solar-zenith-analytical.readthedocs.io/en/fix_see_also_docs_solar_zenith_analytical/generated/pvlib.modelchain.ModelChain.prepare_inputs.html) Assigns attributes list spilling to next line makes the section too confusing. Move it to notes?
* [bifacial](https://fix-see-also-docs-solar-zenith-analytical.readthedocs.io/en/fix_see_also_docs_solar_zenith_analytical/generated/pvlib.bifacial.pvfactors_timeseries.html) parameters not rendered

Things that are entirely broken in both and not required here:
* [SingleAxisTracker.singleaxis](https://fix-see-also-docs-solar-zenith-analytical.readthedocs.io/en/fix_see_also_docs_solar_zenith_analytical/generated/pvlib.tracking.SingleAxisTracker.singleaxis.html)

**Comment by mikofski** [comment]
- [x] `Location.get_solarposition` param _kwargs_
just needed a newline and tab after _kwargs_
image
- [x] new line needed after return type in `solar_zenith_analytical`
image
_FYI_: this is the [expected rtype behavior for sphinx](http://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#info-field-lists), setting [`napoleon_use_rtype = False`](https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html#confval-napoleon_use_rtype) changes this to:
image

- [ ] `Location.get_clearsky` _kwargs_ spec is not sensible with napoleon.
Just needs newline and tab after _kwargs_
image

- [ ] `sapm` returns section
How do you want this to look? Separating rtype is the default Sphinx expected. I'll try `napoleon_use_rtype = False` maybe it looks better:
image

- [ ] `SingleAxisTracker` params not rendered right.
It was missing the **Parameters** heading before the arguments list

- [ ] `ModelChain.prepare_inputs` Assigns attributes list spilling to next line makes the section too confusing. Move it to notes?

- [ ] bifacial parameters not rendered

- [ ] `SingleAxisTracker.singleaxis`

**Comment by wholmgren** [comment]
> FYI: this is the expected rtype behavior for sphinx, setting napoleon_use_rtype = False changes this to:

ah, I prefer the behavior when set to `False`. Your preference?

**Comment by mikofski** [comment]
It appears that the `__init__` method has always been listed in the documentation, see [v0.6.0](https://pvlib-python.readthedocs.io/en/v0.6.0/generated/pvlib.location.Location.html), but I can't figure out how to get rid of it, things I've tried, but failed:
 * get rid of numpydoc extensions
* set numpydoc directive `numpydoc_show_class_members` to false
* set `napoleon_include_init_with_doc` to false
* set `napoleon_include_special_with_doc` to false
* set `napoleon_include_private_with_doc` to false
* delete `source/generated` folder
* hard cache reset the browser
* add extra newline after Parameters section
I give up :(

**Comment by mikofski** [comment]
napolean directives I did not use:

    # napoleon_google_docstring = False
    # napoleon_use_param = False
    # napoleon_use_ivar = True
    # napoleon_include_init_with_doc = False
    # napoleon_include_private_with_doc = False
    # napoleon_include_special_with_doc = False

**Comment by mikofski** [comment]
Make sure to reset browser cache before checking [rendered docs](https://fix-see-also-docs-solar-zenith-analytical.readthedocs.io/en/fix_see_also_docs_solar_zenith_analytical/index.html#)

File: docs/sphinx/source/conf.py [hunk 1]
```diff
@@ -55,12 +55,15 @@ def __getattr__(cls, name):
     'sphinx.ext.viewcode',
     'sphinx.ext.intersphinx',
     'sphinx.ext.extlinks',
-    'numpydoc',
+    'sphinx.ext.napoleon',
     'sphinx.ext.autosummary',
+    'numpydoc',
```
Comment (wholmgren) [inline docs/sphinx/source/conf.py]: delete?

File: pvlib/tracking.py [hunk 2]
```diff
@@ -67,6 +71,21 @@ def __repr__(self):
...
+        ----------
+        apparent_zenith : float, 1d array, or Series
+            Solar apparent zenith angles in decimal degrees.
+
+        apparent_azimuth : float, 1d array, or Series
+            Solar apparent azimuth angles in decimal degrees.
+
+        Returns
+        -------
+        traciking data
```
Comment (wholmgren) [inline pvlib/tracking.py]: typo

File: docs/sphinx/source/conf.py (hunk 1 above)
Comment (mikofski) [reply docs/sphinx/source/conf.py]: It seems to render fine without it, and I suppose we can also remove a dependency, would probably also need to remove the numpydoc directive a few lines above above. Sure - your call. Let's do it. 🚀

Comment (wholmgren) [reply docs/sphinx/source/conf.py]: yes, numpydoc recommends against using it at the same time as napoleon. Also needs to be removed from setup.py

**Pull request merged**